module2/homework/nwdNww/nwdNww.hpp

```diff
@@ -1,11 +1,68 @@
 #pragma once
 
 int NWD(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+    int dividend, divider, result = 1;
+    
+    if(rhs == 0 || lhs == 0)
+    {
+        if(rhs == 0 && lhs != 0) return lhs;
+        else if(rhs != 0 && lhs == 0) return rhs;
+        else return 0;
```

Comment:
```suggestion
        if(rhs == 0 && lhs != 0) {
            return lhs;
        } else if(rhs != 0 && lhs == 0) {
            return rhs;
        }  
        return 0;

```
That's as formatting goes.
But i've seen even better solution: if either lhs or rhs is equal zero, return lhs + rhs.
No extra if statements.